surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/SmartTestingSurefireProvider.java

```diff
@@ -18,16 +17,18 @@
     private ProviderParametersParser paramParser;
     private Class<SurefireProvider> providerClass;
     private ProviderParameters bootParams;
+    private final TestsToRun testsToRun;
+
 
     public SmartTestingSurefireProvider(ProviderParameters bootParams) {
         this.bootParams = bootParams;
         this.paramParser = new ProviderParametersParser(this.bootParams);
         this.providerClass = new ProviderList(this.paramParser).resolve();
         this.surefireProvider = createSurefireProviderInstance();
+        this.testsToRun = (TestsToRun) getSuites();
```

Comment:
I don't think this is the optimal flow of doing things. We can simply check if the flag is set and return `testsToRun` at the moment we get it from the parent (so second line of `getTestsToRun()`) before we kick in all the strategies. 

That would make it simpler to reason about.